extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java

```diff
@@ -52,18 +53,23 @@ public void createExecutionService(BeanContainer beanContainer, Schema schema) {
         return new Handler<RoutingContext>() {
             @Override
             public void handle(RoutingContext event) {
-                if (event.normalisedPath().length() == graphqlUiPath.length()) {
+                if (runtimeConfig.enable) {
+                    if (event.normalisedPath().length() == graphqlUiPath.length()) {
 
-                    event.response().setStatusCode(302);
-                    event.response().headers().set(HttpHeaders.LOCATION, graphqlUiPath + "/");
+                        event.response().setStatusCode(302);
+                        event.response().headers().set(HttpHeaders.LOCATION, graphqlUiPath + "/");
+                        event.response().end();
+                        return;
+                    } else if (event.normalisedPath().length() == graphqlUiPath.length() + 1) {
+                        event.reroute(graphqlUiPath + "/index.html");
+                        return;
+                    }
+
+                    staticHandler.handle(event);
+                } else {
+                    event.response().setStatusCode(404);
```

Comment:
I would simplify this by doing:
```
if (!runtimeConfig.enable) {
    event.response().setStatusCode(404);
    event.response().end();
    return;
}
```
and avoid the `else`.

In general, it's good practice to evacuate the special cases at the beginning to keep the general logic simple.